server/controllers/items-controller.js

```diff
@@ -0,0 +1,22 @@
+const items = require("./../data/items.json");
+
+exports.itemsGet = async (req, res) => {
+  let reqProps = req.body;
+  let retItems = [];
+
+  if (reqProps["merchant-id"] && reqProps["barcode"]) {
+    let queryMerchant = reqProps["merchant-id"];
+    let queryBarcodes = new Set(reqProps["barcode"]);
+    for (let i = 0; i < items.length; ++i) {
```

Comment:
nit: For most of for loops, 
if you do not need to use index i, then instead of using items[i], I would suggest using sth like
for (let item : items) 
Also, consider using filter() in this case as you only want to process the item whose merchant_id is queryMerchant

Reply:
+1